### DISCUSSION TITLE
make supercell issues

### DISCUSSION THREAD

**Discussion by caolz** [mesonepigreco/CellConstructor#92] [Q&A]
I would like to create a supercell with a size of (223), but I encountered the following error.

```python
# This function will use ASE to give us a starting
# guess for dynamical matrices
def get_starting_dynamical_matrices(structure_filename, potential, supercell):
    atoms = ase.io.read(structure_filename)
    atoms.set_calculator(potential)
    ecf = ExpCellFilter(atoms)
    qn = QuasiNewton(ecf)
    qn.run(fmax=0.0005)

    structure = CC.Structure.Structure()
    structure.generate_from_ase_atoms(atoms, get_masses = True)
    dyn = CC.Phonons.compute_phonons_finite_displacements(structure, potential, supercell = supercell)
    dyn.Symmetrize()
    dyn.ForcePositiveDefinite()

    eos = calculate_eos(atoms)
    v0, e0, B = eos.fit()
    bulk = B / kJ * 1.0e24

    return dyn, bulk


# Our input variables
temperature = 300.0
nconf = 5000
max_pop = 1000

# Load in DeepMD potential
pot = DP(model="DPA-Semi-model.pb")
#supercell = tuple((4*np.ones(3, dtype=int)).tolist())
supercell = (2, 2, 3)
dyn, bulk = get_starting_dynamical_matrices('POSCAR-prim', pot, supercell)
```

`Traceback (most recent call last):
  File "/home/ubuntu/user/caolingzhi/1-DPA-Semi/test/sscha-test.py", line 47, in <module>
    dyn, bulk = get_starting_dynamical_matrices('POSCAR-prim', pot, supercell)
  File "/home/ubuntu/user/caolingzhi/1-DPA-Semi/test/sscha-test.py", line 27, in get_starting_dynamical_matrices
    dyn = CC.Phonons.compute_phonons_finite_displacements(structure, potential, supercell = supercell)
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/cellconstructor/Phonons.py", line 4498, in compute_phonons_finite_displacements
    return compute_phonons_finite_displacements_sym(structure, ase_calculator, epsilon,
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/cellconstructor/Phonons.py", line 4857, in compute_phonons_finite_displacements_sym
    correct_dyn.AdjustQStar()
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/cellconstructor/Phonons.py", line 3079, in AdjustQStar
    q_stars, q_order = qe_sym.SetupQStar(self.q_tot)
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/cellconstructor/symmetries.py", line 309, in SetupQStar
    pop_index = np.argmin(q_dist)
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 1325, in argmin
    return _wrapfunc(a, 'argmin', axis=axis, out=out, **kwds)
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 56, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/home/ubuntu/anaconda3/envs/sscha/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 45, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
ValueError: attempt to get argmin of an empty sequence`

**Comment by mesonepigreco** [comment]
Hi, what kind of structure is it? 
It could be that, if it is cubic, the supercell (2,2,3) violates the symmetry and triggers the error in the code. 

Is it the case? Could you provide the structure to help us reproduce the error?

**Reply by caolz** [reply]
I tested the BAs，GaAs and InAs. They all have the same problem.
generated by phonopy
   1.0
    -2.4096269607499998   -2.4096269607499998    0.0000000000000000
    -2.4096269607499998    0.0000000000000000   -2.4096269607499998
     0.0000000000000000   -2.4096269607499998   -2.4096269607499998
As B
   1    1
Direct
  0.0000000000000000  0.0000000000000000  0.0000000000000000
  0.7500000000000000  0.7500000000000000  0.7500000000000000

GaAs
   1.00000000000000
     0.0000000000000000    2.8267950000000000    2.8267950000000000
     2.8267950000000000    0.0000000000000000    2.8267950000000000
     2.8267950000000000    2.8267950000000000    0.0000000000000000
    Ga     As
     1     1
Direct
  0.0000000000000000  0.0000000000000000  0.0000000000000000
  0.7500000000000000  0.7500000000000000  0.7500000000000000

generated by phonopy
   1.0
     0.0000000000000000    3.0291500000000000    3.0291500000000000
     3.0291500000000000    0.0000000000000000    3.0291500000000000
     3.0291500000000000    3.0291500000000000    0.0000000000000000
In As
   1    1
Direct
  0.0000000000000000  0.0000000000000000  0.0000000000000000
  0.7500000000000000  0.7500000000000000  0.7500000000000000

**Comment by mesonepigreco** [comment]
Thanks, it seems indeed that the supercell choosen does not satisfy the same symmetries of the structure. This gives the error when trying to enforce the symmetrization collecting the q points of the 2x2x3 mesh in stars. I assume that both the 2x2x2 and 3x3x3 supercells work, do they?

**Comment by caolz** [comment]
When 2x2x2 and 3x3x3 are fine.
